components/feature/session/src/main/java/mozilla/components/feature/session/SessionIntentProcessor.kt

```diff
@@ -43,22 +51,42 @@ class SessionIntentProcessor(
                 true
             }
 
+            action == Intent.ACTION_SEND -> {
+                handleActionSend(url)
+                true
+            }
             else -> {
-                val session = if (openNewTab) {
-                    Session(url, false, Source.ACTION_VIEW).also { sessionManager.add(it, selected = true) }
-                } else {
-                    sessionManager.selectedSession ?: Session(url, false, Source.ACTION_VIEW)
-                }
-
+                val session = createSession(url)
```

Comment:
Would be nicer to have a `handleActionView` method here as well, similar to `handleActionSend`

Reply:
👍